### DISCUSSION TITLE
Inclusion of derivatives?

### DISCUSSION THREAD

**Discussion by ethanj20** [skfolio/skfolio#44] [General]
Just checking out the repo - looks extremely interesting and I look forward to diving in. Any plans to consider financial derivatives in the optimization models?

**Comment by HugoDelatte** [comment]
Thank you!
The optimization models are agnostic to the type of underlyings as they simply consume the returns distribution from the prior model.
However, the current implementation of the moment estimators assumes that the input `X` is a market invariant. 
We plan to implement specific estimators for Bonds and Options but we don't have an ETA yet.
In the meantime, you will have to implement a custom moment estimator to do that.
Have a looks at the [data preparation guide](https://skfolio.org/user_guide/data_preparation.html) and the references from Meucci.
Also, don't hesitate to submit a PR if you implement an elegant solution.
